### DISCUSSION TITLE
想了解一下各加速方式的区别

### DISCUSSION THREAD

**Discussion by Kulahala** [BeyondDimension/SteamTools#1725] [Q&A] [answered]
一直用的都是host方式，想问下PAC和DNS驱动拦截都有什么区别

**Comment by ElemenTP** [comment]
核心区别在如何将软件的流量转发到本地steam++的代理服务器中，host和DNS驱动拦截都是在域名解析的时候直接指向代理服务器，PAC则是使用一种代理设置脚本，让流量走http代理的方式发送到代理服务器。

**Reply by Kulahala** [reply]
那么一般那种方式效果最好？

**Reply by bangkibang** [reply]
同想知道这个四种模式的区别，尤其是在并行使用CLASH和V2ray的时候，哪种比较合理高效

**Comment by willbeetter** [comment]
个人感觉Hosts模式好（不了解，勿喷）

**Reply by Kulahala** [reply]
我也是一直用着host，想了解其他

**Comment by CDmking** [comment, marked as answer]
此评论需要了解“hosts文件”的原理、“DNS”的原理、“代理服务器”、“IP地址”等知识后查看。
此加速器原理是将本地电脑作为代理服务器与网站进行交流，所以只要有方法能将流量通过加速器程序传输就能达到加速的效果，即将流量先通过本地电脑（localhost，IP地址为“0.0.0.0”或“127开头的地址”）进行代理再传输给网站。
- “Hosts代理模式”是将需要加速的网站的网址填入hosts文件，IP地址均为127.0.0.1或0.0.0.0，代理的网站仅正在加速的网站；速度较快，但程序需要获取修改hosts文件的权限。
- “系统代理模式”与“PAC代理模式”是将系统的代理服务器地址设置为127.0.0.1，代理的网站为所有网站；由于技术原因，速度较慢。
- “DNS驱动拦截模式”是将本地电脑发给DNS的信息拦截下来，并将网站服务器IP地址修改为127.0.0.1而实现代理，代理的网站仅为正在加速的网站；速度较快，仅Windows平台可用。

**Reply by bangkibang** [reply]
不具备相关专业知识，仅从基本使用体验来说，开启windows端绕过大陆/白名单模式的代理工具时(clash)，建议选择以上哪种加速方式？感谢

**Reply by CDmking** [reply]
若只使用加速功能，只建议使用“Hosts代理模式”。

**Reply by cylixlee** [reply]
请问使用HOSTS模式访问Github和自己从第三方（如GithubHosts）获取hosts文件并替换系统hosts有什么区别呢，谢谢🙏

**Reply by CDmking** [reply]
当hosts文件指向的IP为127.0.0.1或0.0.0.0时，由加速器代理，数据传输过程：浏览器 <---> Steam++ <-- 服务器；IP不为127.0.0.1或0.0.0.0时，为直接访问，相当于跳过DNS，数据传输过程：浏览器 --> 服务器 --> 浏览器。
注意：上面的数据传输过程是最简化的，实际传输过程要复杂的多，而且实际上还有其他因素的干扰（如本地电脑到服务器的物理距离、国家防火墙等），导致反向代理工具（如Steam++）要比直接访问速度更快，也能访问一些平时访问不到的服务器。

**Comment by elarbor** [comment]
咱也不懂，就是能打开github就行😂
